pkg/kubelet/cm/memorymanager/policy_static.go

```diff
@@ -97,29 +98,328 @@ func (p *staticPolicy) Start(ctx context.Context, s state.State) error {
+			hasSharedAppContainers = true
+			break
+		}
+	}
+	if !hasSharedAppContainers {
+		for _, container := range pod.Spec.InitContainers {
+			if !podutil.IsRestartableInitContainer(&container) {
+				continue
+			}
+			req, err := getRequestedResources(ctx, pod, &container)
+			if err != nil {
+				return err
+			}
+			if len(req) == 0 {
+				hasSharedAppContainers = true
+				break
+			}
+		}
+	}
+
+	if hasSharedAppContainers {
+		for resourceName, totalPodResource := range podTotalMemory {
+			if sumOfLongRunningExclusiveMemory[resourceName] >= totalPodResource {
+				return fmt.Errorf("pod rejected, sum of exclusive container %s requests equals pod budget, leaving no memory for shared containers", resourceName)
+			}
+		}
+	}
+
+	// Check for empty shared pool for standard init containers.
+	sumOfSidecarExclusiveMemory := make(map[v1.ResourceName]uint64)
+	for _, container := range pod.Spec.InitContainers {
+		if podutil.IsRestartableInitContainer(&container) {
+			req, err := getRequestedResources(ctx, pod, &container)
+			if err != nil {
+				return err
+			}
+			for resourceName, val := range req {
+				sumOfSidecarExclusiveMemory[resourceName] += val
+			}
+		}
+	}
+
+	poolForStandardInits := make(map[v1.ResourceName]uint64)
+	for rn, val := range podTotalMemory {
+		poolForStandardInits[rn] = val - sumOfSidecarExclusiveMemory[rn]
+	}
+
+	maxStandardInitExclusiveMemory := make(map[v1.ResourceName]uint64)
+	hasSharedStandardInitContainer := false
+	for _, container := range pod.Spec.InitContainers {
+		if !podutil.IsRestartableInitContainer(&container) {
+			req, err := getRequestedResources(ctx, pod, &container)
+			if err != nil {
+				return err
+			}
+			if len(req) > 0 {
+				for resourceName, val := range req {
+					if val > maxStandardInitExclusiveMemory[resourceName] {
+						maxStandardInitExclusiveMemory[resourceName] = val
+					}
+				}
+			} else {
+				hasSharedStandardInitContainer = true
+			}
+		}
+	}
+
+	for resourceName, poolSize := range poolForStandardInits {
+		if maxStandardInitExclusiveMemory[resourceName] > poolSize {
+			return fmt.Errorf("pod rejected, largest exclusive init container requests %d of %s, but only %d available", maxStandardInitExclusiveMemory[resourceName], resourceName, poolSize)
+		}
+	}
+
+	if hasSharedStandardInitContainer {
+		for resourceName, size := range poolForStandardInits {
+			if size <= 0 {
+				return fmt.Errorf("pod rejected, pod has shared init containers but no %s available for them", resourceName)
+			}
+		}
+	}
+
+	return nil
+}
+
+// AllocatePod is the entry point for pod-level resource allocation.
+// It's called once per pod by the Topology Manager's pod-scope admit handler.
+// The logic here allocates a single NUMA-aligned "bubble" of memory for the
+// entire pod. All containers within the pod will share this NUMA binding.
+func (p *staticPolicy) AllocatePod(ctx context.Context, s state.State, pod *v1.Pod, hint topologymanager.TopologyHint) (rerr error) {
```

Comment:
likewise for cpumanager, we need as complete as possible *unit* test coverage and as much as possible e2e coverage for the main flows especially considering the desired aggressive timeline for beta graduation (xref https://github.com/kubernetes/enhancements/pull/5242)

Reply:
Added the unit test for `AllocatePod` in both managers, however, same case as E2E tests, still need to add more test cases.